src/lib/registration/ui/addon_selection_base_dialog.rb

```diff
@@ -71,7 +71,13 @@ def addon_widget_id(addon)
       # @param [Boolean] enable true for filtering beta releases
       def filter_beta_releases(enable)
         self.class.filter_beta = enable
-        @addons = enable ? @all_addons.reject(&:beta_release?) : @all_addons
+        if enable
+          unregistered_addons = @all_addons.reject(&:registered?)
+          @addons = @all_addons.select(&:registered?)
```

Comment:
btw as you split by registered, you can have just one call - http://apidock.com/ruby/Enumerable/partition
so something like
```
@addons, available_addons = @all_addons.partition(&:registered?)
```